src/main/java/hudson/plugins/mercurial/MercurialSCMSource.java

```diff
@@ -195,22 +195,19 @@ public void setCredentialsId(@CheckForNull String credentialsId) {
                 .newRequest(this, listener) ) {
             MercurialInstallation inst = MercurialSCM.findInstallation(request.installation());
             if (inst == null) {
-                listener.error("No configured Mercurial installation");
-                return;
+                throw new IOException("No configured Mercurial installation");
```

Comment:
```suggestion
                throw new AbortException("No configured Mercurial installation");
```

and similarly elsewhere.